wunderio/next4drupal-project
# NEX-44: Add working webform example

This branch adds a simple working webform example in backend and frontend.

We decided (for now at least) to implement a static form, in the sense that the "hardcoded" fields in the frontend need to match the fields in the webform in the backend. Creating a dynamic system of forms that automatically match what's coming from the user-defined webforms in Drupal is out of scope for this template.

On the backend:
* Added webform and webform_rest
* added permission for anonymous user to post to webform

On the frontend:
* added an API route for the form to post to, which in turn will post to drupal
* added a form component with fields matching the fields defined in the default contact webform on the drupal side.
* Added translations for the user interface of the form
* Added hook_react_form to be able to reset the form and better handle it. I have not added validation since the html5 validation handles this basic case. I have added a thank you message with a button to submit the form again.

### Testing instructions

1. run the megacommand to setup the project from scratch https://github.com/wunderio/next4drupal-project#%EF%B8%8F-quick-one-command-setup
2.  visit frontpage, submit form, see thank you message, click button to submit again, see form again
3.  log into backend `lando drush uli`
4.  visit backend at: https://next4drupal-project.lndo.site/en/admin/structure/webform/manage/contact/results/submissions  and you should find your submission 

### Screenshots
Form on frontpage:
![image](https://user-images.githubusercontent.com/185412/221505592-c3043f2c-589c-4514-a8be-cf01bdf901bb.png)

Confirmation message + button to reset form:
![image](https://user-images.githubusercontent.com/185412/221505644-078133b8-ed84-49d4-8e68-c9986b1dc196.png)

Submission data in backend:
![image](https://user-images.githubusercontent.com/185412/221505753-d660bb56-6db4-4c8e-8ec8-f315d906a36d.png)
